packages/fx-core/tests/plugins/resource/frontend/unit/generateArmTemplates.test.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import "mocha";
+
+import * as chai from "chai";
+import chaiAsPromised from "chai-as-promised";
+import fs from "fs-extra";
+import * as path from "path";
+
+import { AzureSolutionSettings, PluginContext } from "@microsoft/teamsfx-api";
+import { TestHelper } from "../helper";
+import { FrontendPlugin } from "../../../../../src";
+import { mockSolutionUpdateArmTemplates } from "../../util";
+import {
+  DependentPluginInfo,
+  FrontendPluginInfo,
+} from "../../../../../src/plugins/resource/frontend/constants";
+
+chai.use(chaiAsPromised);
+
+describe("FrontendGenerateArmTemplates", () => {
+  let frontendPlugin: FrontendPlugin;
+
+  beforeEach(() => {
+    frontendPlugin = new FrontendPlugin();
+  });
+
+  it("generate bicep arm templates", async () => {
+    // Act
+    const activeResourcePlugins = [
+      FrontendPluginInfo.PluginName,
+      DependentPluginInfo.AADPluginName,
+      DependentPluginInfo.RuntimePluginName,
+    ];
+    const pluginContext: PluginContext = TestHelper.getFakePluginContext();
+    pluginContext.projectSettings!.solutionSettings = {
+      name: "test_solution",
+      version: "1.0.0",
+      activeResourcePlugins: activeResourcePlugins,
+    } as AzureSolutionSettings;
+    const result = await frontendPlugin.generateArmTemplates(pluginContext);
+
+    // Assert
+    const testModuleFileName = "frontend_hosting_test.bicep";
+    const mockedSolutionDataContext = {
+      plugins: activeResourcePlugins,
+      "fx-resource-frontend-hosting": {
+        modules: {
+          frontendHostingProvision: {
+            path: `./${testModuleFileName}`,
+          },
+        },
+      },
+    };
+    chai.assert.isTrue(result.isOk());
+    if (result.isOk()) {
+      const expectedResult = mockSolutionUpdateArmTemplates(
+        mockedSolutionDataContext,
+        result.value
+      );
+
+      const expectedBicepFileDirectory = path.join(__dirname, "expectedBicepFiles");
```

Comment:
Seems the ut generate a series output files. Do we need to clean up those after ut done?

Reply:
NIT: And I think the best practice is unit test should not touch file system. You could consider using [mock-fs](https://www.npmjs.com/package/mock-fs) though it significantly increases your work...